.circleci/scripts/det_deploy.sh

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+det deploy $@
+status=$?
+# if we tried to bring up a cluster and it failed, recover by trying to bring the cluster down
+if [ $status -ne 0 ] && [ "$2" == "up" ] && [ "${CLUSTER_ID}" != "" ]; then
```

Comment:
Does `${CLUSTER_ID}` exist automatically? I thought you needed an `environment:` block to get parameters as env vars. 🤷‍♂️

Reply:
it looks like this works because the `set-cluster-id` step is called during `setup-aws-cluster`. It gets confusing because `deploy` will call `deploy-aws-cluster` without the `set-cluster-id` step, which is what we want because in the `deploy-latest-master-cluster` case it should not try to clean up on failure.